File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        
        // for(int i=0;i<nums.size();++i)
        //     for(int j=i;j<nums.size();++j)
        //         for(int k=j;k<nums.size();++k)
        //             if(i<j<k&&nums[i]<nums[j]&&nums[j]<nums[k])
        //                 return true;
        // return false;
        
         int small =INT_MAX, big = INT_MAX;
        for (int n : nums) {
            if (n <= small) 
              small = n; 
            else if (n <= big)
               big = n;  
            else 
                  return true;
        }
        return false;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        small,big=10000000000,10000000000
        for n in nums:
            if n <= small: 
                small = n
            elif n <= big:
                big = n 
            else:
                return True
        return False